higher-order.js

```diff
@@ -67,18 +84,52 @@ const _ = {
      * @returns {Array} filteredArray - The filtered array.
      */
     filter: function(arr, filterApplier) {
-        if (typeof filterApplier !== 'function') {
-            throw new TypeError('The filter applier must be a valid closure.');
-        }
-
+        guardFunction(filterApplier);
+        
         return this.reduce(arr, (previous, element) => {
-            if (filterApplier(element) === true) {
+            if (filterApplier(element)) {
                 previous.push(element);
             }
 
             return previous;
         }, []);
+    },
+
+    /**
+     * Given an array and a function, runs the given function on every element and returns true if all elements are accounted.
+     * @param {Array} arr - The array to be checked.
+     * @param {Function} accounter - The function in charge of checking if a value should be considered.
+     * @returns {Boolean} - The result of applying accounter on every element.
+     */
+    all: function(arr, accounter) {
+        guardArray(arr);
+        
+        if (arr.length === 0) {
+            return false;
```

Comment:
Careful!  You want `_.all([], fn)` to return `true` for any function `fn`.  This is why you pass in `true` as the initial value for `reduce` below.  We want the following law/rule/equation to hold for all lists and all functions:

```javascript
_.all(listA + listB, fn) === _.all(listA, fn) && _.all(listB, fn)
```

This forces us:

```javascript
_.all(listB, fn) === _.all([] + listB, fn)              // since listB + [] === listB
                 === _.all([], fn) && _.all(listB, fn)  // since we want the above rule to hold
```

And this means that `_.all([], fn)` *must* return `true`.

This corresponds to the concept of [vacuous truth](https://en.wikipedia.org/wiki/Vacuous_truth) in logic ([Portuguese version](https://pt.wikipedia.org/wiki/Verdade_por_vacuidade)).